activerecord/test/cases/calculations_test.rb

```diff
@@ -358,6 +358,12 @@ def test_count_with_distinct
     assert_equal 4, Account.select(:credit_limit).uniq.count
   end
 
+  def test_count_with_distinct_emits_single_distinct_in_query
+    assert_equal 4, Account.select(:credit_limit).distinct.count
+    queries = assert_sql { Account.select(:credit_limit).distinct.count}
+    assert_equal 1, queries.last.scan(/DISTINCT/).count
```

Comment:
How about we test the expected result sets, instead of making vague assertions about the generated SQL?